src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php

```diff
@@ -107,22 +107,6 @@ protected function getModel()
     }
 
     /**
-     * validate and save model after update or insertion.
-     * Use the reference node and tag to rename validation output for a specific node to a new offset, which makes
-     * it easier to reference specific uuids without having to use them in the frontend descriptions.
-     * @param $node reference node, to use as relative offset
-     * @return array result / validation output
-     */
-    protected function validateAndSave($node = null)
```

Comment:
I don't agree here, if you remove an item you don't have the ability to avoid the hook on save for example.
This https://github.com/opnsense/core/blob/master/src/opnsense/mvc/app/controllers/OPNsense/IDS/Api/SettingsController.php#L330-L331 could be replaced by validateAndSave in the new construct.